bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java

```diff
@@ -124,14 +127,28 @@ private boolean verifyLedgerDirs(Cookie c, boolean checkIfSuperSet) {
         }
     }
 
-    private void verifyInternal(Cookie c, boolean checkIfSuperSet) throws BookieException.InvalidCookieException {
+    private boolean verifyIndexDirs(Cookie c, boolean checkIfSuperSet, boolean enforceCookieIndexDirCheck) {
+        if (!enforceCookieIndexDirCheck) {
+            return true;
+        }
+
+        if (!checkIfSuperSet) {
+            return indexDirs.equals(c.indexDirs);
+        } else {
+            return isSuperSet(decodeDirPathFromCookie(indexDirs), decodeDirPathFromCookie(c.indexDirs));
+        }
+    }
+
+    private void verifyInternal(Cookie c, boolean checkIfSuperSet, ServerConfiguration conf) throws BookieException.InvalidCookieException {
```

Comment:
can't we simply pass 'conf.isEnforceCookieIndexDirCheck' instead of the full object ?
why did you choose to pass the reference to the full ServerConfiguration object ?

Reply:
I passed a conf object for future cookie changes that we bring in, viz. network location for 3AZ work we are doing. 
A conf object would make it easier than to refactor the function definition to accept yet another boolean.